configure.ac

```diff
@@ -32,6 +32,7 @@ AH_BOTTOM([
 dnl AX_CHECK_ENABLE_DEBUG must be called before AC_PROG_CC or
 dnl AM_PROG_AR
 AX_CHECK_ENABLE_DEBUG([no])
+AC_USE_SYSTEM_EXTENSIONS
 dnl AM_PROG_AR must be called before LT_INIT
```

Comment:
Add some rationale, like we acknowledge that we're using a number of linux-isms at this point, so the risk of silently enabling system extensions is minimal.  Basically, leave a note for the next guy.

Reply:
sorry to be pedantic but to be clear this specific PR [will work on fbsd](https://man.freebsd.org/cgi/man.cgi?query=memfd_create&manpath=FreeBSD+15.0-CURRENT) and [musl](http://git.musl-libc.org/cgit/musl/tree/src/linux/memfd_create.c). What we're admitting here is just that the codebase is not ANSI C, but that's not the same thing as requiring glibc or linux.